application/helpers/questionHelper.php

```diff
@@ -1638,15 +1638,21 @@ public static function getQuestionThemeAttributeValues($sQuestionThemeName = nul
         $xml_config = is_file($sCoreThemeXmlPath) ? simplexml_load_file($sCoreThemeXmlPath) :  simplexml_load_file($sUserThemeXmlPath);
         $custom_attributes = json_decode(json_encode((array)$xml_config->custom_attributes), TRUE);
         libxml_disable_entity_loader(true);
-        // no need recursive QuestionAttribute::getDefaultSettings() is array of string or null
-        $defaultQuestionAttributeValues = QuestionAttribute::getDefaultSettings();
-        
-        $additionalAttributes = array_map(function($arr) use ($defaultQuestionAttributeValues){
-            return array_merge($defaultQuestionAttributeValues, $arr);
-        },$custom_attributes['attribute']);
 
+        if(!empty($custom_attributes['attribute']['name'])) {
+            // Only one attribute set in config : need an array of attributes
+            $custom_attributes['attribute'] = array($custom_attributes['attribute']);
+        }
+
+        $defaultQuestionAttributeValues = QuestionAttribute::getDefaultSettings();
+        $additionalAttributes = array();
+        // Create array of attribute with name as key
+        foreach($custom_attributes['attribute'] as $customAttribute) {
+            if(!empty($customAttribute['name'])) {
```

Comment:
This allow to add more control if needed (and don't find an array_map system to update key (find but … a foreach is more understandable here)